test/index.js

```diff
@@ -158,3 +158,60 @@ t.test('packs from registry spec', async t => {
     'packed directory matches expectations'
   )
 })
+
+t.test('files packed are alphabetically sorted', async (t) => {
+  const testDir = t.testdir({
+    index: 'index',
+    dist: {
+      'foo.js': 'foo',
+      'foo.ts': 'foo',
+      bar: 'bar',
+      baz: 'baz'
+    },
+    numbers: {
+      1: '1',
+      11: '1',
+      '01': '0',
+      20: '2',
+      2: '2',
+      10: '1',
+      0: '0'
+    },
+    'package.json': JSON.stringify({
+      name: 'pkg'
+    }),
+    lib: {
+      'index.js': 'lib/index',
+      'foo.js': 'lib/foo'
+    },
+    'README.md': 'readme-1',
+    'readme.txt': 'readme-2'
+  })
+
+  const target = `${testDir}/my-cool-pkg-1.0.0.tgz`
+  const tarContents = await pack(testDir, { target })
+  const tarFilePaths = tarContents.files.map(file => file.path)
+  const expectedOrder = [
+    'dist/bar',
+    'dist/baz',
+    'dist/foo.js',
+    'dist/foo.ts',
+    'index',
+    'lib/foo.js',
+    'lib/index.js',
+    'numbers/0',
+    'numbers/01',
+    'numbers/1',
+    'numbers/2',
+    'numbers/10',
+    'numbers/11',
+    'numbers/20',
+    'package.json',
+    'README.md',
```

Comment:
hi @eridal thanks for the contribution! this looks great ❤️ 

now one very small change I'd like to ask - I know the GitHub web interface order files in such a way that capitalized letters come first (I _think_ it comes from some sort of unix standard but I'm really not sure about it, don't quote me on that heheheh) anyways, the point is that's a standard that people got used to and it would be nice if we could replicate the same thing here 😊

Reply:
@ruyadorno thanks for your feedback

Yes, at first that was the order I had set in place. Then I thought that given different OSes have different standards (ie: case vs icase, folders first, ...) the more inclusive approach would be to sort naturally.

But now that you mentioned, using the same listing order than Github really makes sense to me!

Reply:
Hmm I did not like much the way it ended up being, but if I'm reading correctly the standard for `localeCompare` we cannot sort this as a single operation.

What I'm seeing is that using **caseFirst** with values **upper** or **lower** will yield orders `AaBbCcDd..` and `aAbBcCdD...` respectively.

Given that we want `ABCD..XYZabc...xyz` then the only option (that I see) is to solve this in a two-steps process.

Please let me know your thoughts about this

> https://www.ecma-international.org/ecma-402/1.0/#CompareStrings